plugin/src/main/scala/com/lightbend/sbt/JavaFormatterPlugin.scala

```diff
@@ -93,12 +91,14 @@ object JavaFormatterPlugin extends AutoPlugin {
       includeFilter in format := "*.java",
       sourceLevel := None,
       targetLevel := None,
-      javaFormattingSettingsFile := None,
-      javaFormattingSettingsFile := javaFormattingSettingsFile.value,
+      javaFormattingSettingsFile in ThisBuild := Some(baseDirectory.value / "formatting-java.xml"),
```

Comment:
Won't this re-parse the same XML file over and over again? There was a reason that the `settings` where in a `build` setting and not a `project` setting, since XML parsing is a huge memory hog and caused noticeable slowdowns for our build.

```scala
  override def buildSettings = Seq(
    // Don't parse the XML file for every single project over and over again!
    JavaFormatterKeys.settings in ThisBuild :=
      JavaFormatterPlugin.settingsFromProfile((baseDirectory in ThisBuild).value / "project" / "java-formatting.xml")
  )
```